usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -6882,10 +6882,10 @@ l2arc_write_buffers(spa_t *spa, l2arc_dev_t *dev, uint64_t target_sz)
 
 	ASSERT3U(write_asize, <=, target_sz);
 	ARCSTAT_BUMP(arcstat_l2_writes_sent);
-	ARCSTAT_INCR(arcstat_l2_write_bytes, write_asize);
+	ARCSTAT_INCR(arcstat_l2_write_bytes, write_psize);
 	ARCSTAT_INCR(arcstat_l2_size, write_sz);
-	ARCSTAT_INCR(arcstat_l2_asize, write_asize);
-	vdev_space_update(dev->l2ad_vdev, write_asize, 0, 0);
+	ARCSTAT_INCR(arcstat_l2_asize, write_psize);
```

Comment:
Actually, I must withdraw my previous comment. There still will be a distinction between psize and asize.. And thus far we seem to account arcstat_l2_asize, the vdev space and l2ad_alloc refcount interms of pszie. So, I am going to follow your advice and rename arcstat_l2_asize to arcstat_l2_psize. Maybe later on we will change how the accounting is done.  E.g. I think that the vdev space should be accounted in terms of asize.  But that's an issue for another day (I hope).